### DISCUSSION TITLE
Get Relationship data before update

### DISCUSSION THREAD

**Discussion by Professor93M** [filamentphp/filament#8615] [Help] [answered]
Dears,

how to get relationship data in the **Edit Page**, because if I used **handleRecordUpdate** or **mutateFormDataBeforeSave** I can't catch the the relation data, and it will be already updated after these methods.

What I want to do is:
**before updating the data I want to check the old data and the new data and compare them before update.**
So how I can get the relationship data before it is updated to handle it?

Thanks.

**Comment by leandrocfe** [comment]
I think you can set `->dehydrated(true)` in your relationship field and use [handleRecordUpdate](https://filamentphp.com/docs/3.x/panels/resources/editing-records#customizing-the-saving-process) on your Edit Page to get the record, data and customize the process

```php
protected function handleRecordUpdate(Model $record, array $data): Model
{
    //now you can access $data['yourRelationship']...
    $record->update($data);
    return $record;
}
```

**Comment by Professor93M** [comment]
thank you for your reply,

But my problem is still not fixed, the problem is that I want to catch the relation data before updating.
because after I dd the $data in this method it returns the updated data for the relationship.
and I want the old relation data before update.

**Reply by leandrocfe** [reply, marked as answer]
I think you can use `saveRelationshipsUsing()` in your relationship

```php
->saveRelationshipsUsing(static function ($component, Model $record, $state) {
    
    //$record->yourRelationship  original data
    //$state current data
    
    $newState = [1, 2, 3];

    $relationship = $component->getRelationship();
    $relationship->sync($newState); //associate, attach, etc..
})
```

**Reply by jwhoami** [reply]
It seems to me that the saveRelationshipUsing is called first and then the handleRecordUpate.

However, is it possible to prevent handleRecordUpdate from running in case saveRelationshipUsing fails
due to some business logic?

Thanks

**Comment by kpsanmao** [comment]
Problem solved?
